setup.py

```diff
@@ -1,166 +1,16 @@
-        return linux_scripts + [script + ".bat" for script in linux_scripts]
-    return linux_scripts
-
-
-def get_packages(directory, prefix):
-    """return a list of subpackages for the given directory"""
-    result = []
-    for package in os.listdir(directory):
-        absfile = join(directory, package)
-        if isdir(absfile):
-            if exists(join(absfile, "__init__.py")):
-                if prefix:
-                    result.append(f"{prefix}.{package}")
-                else:
-                    result.append(package)
-                result += get_packages(absfile, result[-1])
-    return result
-
-
-def _filter_tests(files):
-    testdir = join("pylint", "test")
-    return [f for f in files if testdir not in f]
-
-
-if easy_install_lib:
-
-    class easy_install(easy_install_lib.easy_install):
-        # override this since pip/easy_install attempt to byte compile
-        # test data files, some of them being syntactically wrong by design,
-        # and this scares the end-user
-        def byte_compile(self, files):
-            files = _filter_tests(files)
-            easy_install_lib.easy_install.byte_compile(self, files)
-
-
-def install(**kwargs):
-    """setup entry point"""
-    if USE_SETUPTOOLS:
-        if "--force-manifest" in sys.argv:
-            sys.argv.remove("--force-manifest")
-    packages = ["pylint"] + get_packages(join(base_dir, "pylint"), "pylint")
-    if USE_SETUPTOOLS:
-        if install_requires:
-            kwargs["install_requires"] = install_requires
-            kwargs["dependency_links"] = dependency_links
-        kwargs["entry_points"] = {
-            "console_scripts": [
-                "pylint = pylint:run_pylint",
-                "epylint = pylint:run_epylint",
-                "pyreverse = pylint:run_pyreverse",
-                "symilar = pylint:run_symilar",
-            ]
-        }
-    kwargs["packages"] = packages
-    cmdclass = {"build_py": build_py}
-    if easy_install_lib:
-        cmdclass["easy_install"] = easy_install
-    return setup(
-        name="pylint",
-        version=__pkginfo__["version"],
-        license=__pkginfo__["license"],
-        description=__pkginfo__["description"],
-        long_description=long_description,
-        author=__pkginfo__["author"],
-        author_email=__pkginfo__["author_email"],
-        url=__pkginfo__["web"],
-        scripts=ensure_scripts(scripts),
-        classifiers=__pkginfo__["classifiers"],
-        data_files=data_files,
-        ext_modules=ext_modules,
-        cmdclass=cmdclass,
-        extras_require=extras_require,
-        test_suite="test",
-        python_requires="~=3.6",
-        setup_requires=pytest_runner,
-        tests_require=["pytest", "pytest-benchmark"],
-        project_urls=project_urls,
-        **kwargs,
-    )
-
+with open(HERE / "pylint/__pkginfo__.py", encoding="UTF-8") as f:
+    exec(f.read(), __pkginfo__)  # pylint: disable=exec-used
 
 if __name__ == "__main__":
```

Comment:
I don't think its necessary here. `setup.py` is only ever executed to build the packages.
The default I'm used to is
```py
from setuptools import setup

setup(
  ...
)
```

Reply:
A test for python 3.8 and 3.9 was failing without it (benchmark I think)

Reply:
Probably a bad test then 😄

Reply:
It's ``test_baseline_benchmark_j1_all_checks_lots_of_files``, I have no idea why it fails though. 
```python
        try:
            ok = dist.parse_command_line()
        except DistutilsArgError as msg:
>           raise SystemExit(gen_usage(dist.script_name) + "\nerror: %s" % msg)
E           SystemExit: usage: __main__.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
E              or: __main__.py --help [cmd1 cmd2 ...]
E              or: __main__.py --help-commands
E              or: __main__.py cmd --help
E           
E           error: no commands supplied
```